org.adempiere.pipo/src/org/adempiere/pipo2/PoFiller.java

```diff
@@ -408,7 +412,33 @@ private void setBlob(String qName) {
 						throw new AdempiereException(e.getLocalizedMessage(), e);
 					}
 				}
-				po.set_ValueNoCheck(qName, data);
+				if ("BinaryData".equals(qName) && data instanceof byte[]) {
+					if (po instanceof MArchive) {
+						/* it comes as a zip file with a single PDF file */
+					    ZipInputStream zipStream = new ZipInputStream(new ByteArrayInputStream((byte[]) data));
+					    byte[] output = null;
+						try {

```

Comment:
should use try(ZipInputStream zipStream = new ZipInputStream(new ByteArrayInputStream((byte[]) data));) and we can remove the zipstream.close call below.